app/services/submission/plant_trial_finalizer.rb

```diff
@@ -35,10 +35,80 @@ def create_new_trait_descriptors
 
   def create_scoring
     trait_mapping = submission.content.step04.trait_mapping
+    trait_scores = submission.content.step04.trait_scores || {}
+    accessions = submission.content.step04.accessions
+    replicate_numbers = submission.content.step04.replicate_numbers || {}
+    design_factors = submission.content.step04.design_factors || {}
+    design_factor_names = submission.content.step04.design_factor_names || []
+    lines_or_varieties = submission.content.step04.lines_or_varieties || {}
+
+    @new_plant_scoring_units = trait_scores.map do |plant_id, scores|
+      rollback(2) unless accessions[plant_id] && accessions[plant_id]['plant_accession'] && accessions[plant_id]['originating_organisation']
```

Comment:
How is that possible? Doesn't parser ignore lines with missing accession? In case it is due to a user error - will user be informed explicitly what was wrong? If that's a guard for really unexpected situation then we should probably just fail.